### PR TITLE
iDeal Component for SFRA

### PR DESCRIPTION
**Pull request by maassenbas**
- iDeal component with version 2.1.0
- Added extra logging for credit card transactions

### CODE REVIEW

File: cartridges/int_adyen_SFRA/cartridge/client/default/js/adyenCheckout.js [hunk 1]
```diff
@@ -146,69 +151,89 @@
...
-                });
-                additionalFields.append(issuers);
-                li.append(additionalFields);
-            }
+        if (paymentMethod.type == "ideal") {
+            var idealContainer = document.createElement("div");
+            $(idealContainer).addClass('hppAdditionalFields').attr('id', 'component_' + paymentMethod.type).attr('style', 'display:none');
+            idealComponent = checkout.create('ideal', {
+                items: paymentMethod.details[0].items,
+                onChange: function (state) {
```
Comment (cyattilakiss) [inline cartridges/int_adyen_SFRA/cartridge/client/default/js/adyenCheckout.js]: remove if not used

File: cartridges/int_adyen_SFRA/cartridge/client/default/js/adyenCheckout.js [hunk 2]
```diff
@@ -146,69 +151,89 @@
...
+                },
+                onValid: function (state, component) {
+                    $('#selectedIssuer').val(state.data.issuer);
+                    $('#adyenIssuerName').val(component.componentRef.props.items.find(x => x.id == state.data.issuer).name);
+                },
+                onError: function (state) {
```
Comment (cyattilakiss) [inline cartridges/int_adyen_SFRA/cartridge/client/default/js/adyenCheckout.js]: same here

File: cartridges/int_adyen_overlay/cartridge/scripts/util/AdyenHelper.ds [hunk 3]
```diff
@@ -152,7 +152,7 @@ var __AdyenHelper : Object = {
     },
 
     getCheckoutUrl : function () : String {
-        var checkoutUrl = this.getLoadingContext() + "sdk/2.0.0/adyen.js";
+        var checkoutUrl = this.getLoadingContext() + "sdk/2.1.0/adyen.js";
```
Comment (cyattilakiss) [inline cartridges/int_adyen_overlay/cartridge/scripts/util/AdyenHelper.ds]: Just double checking if you have done all the necessary changes moving to v. 2.1.0 based on the list I've sent you

File: cartridges/int_adyen_overlay/cartridge/scripts/util/AdyenHelper.ds [hunk 4]
```diff
@@ -283,31 +283,42 @@ var __AdyenHelper : Object = {
 
     createCardObjectSF: function (isOneClick, formType) {
         var card;
+        var cardType = "scheme";
```
Comment (rikterbeek) [inline cartridges/int_adyen_overlay/cartridge/scripts/util/AdyenHelper.ds]: see below some changes regarding cards could you explain what it does and why it included into the ideal component PR ?

**Pull request merged**